### DISCUSSION TITLE
Connect Postgres service with custom container image

### DISCUSSION THREAD

**Discussion by everspader** [community/community#25693] [Actions]
<p>In my Django project, I have a CI workflow for running tests, which requires a Postgres service. Recently a new app introduced heavier packages such as pandas, matplotlib, pytorch and so on and this increased the <code>run-tests</code> job time from 2 to 12 minutes which is absurd. Also in my project, I have a base Docker image with Python and these packages that are heavier to speed up the build of the images. So I was thinking to use this same image in the workflow when running the steps because the packages would be loaded already.</p>
<p>Unfortunately, all goes well until it reaches the step to actually run the tests because it seems that the postgres service is not connected with the container and I get the following error:</p>
<pre><code class="lang-auto">psycopg2.OperationalError: could not connect to server: Connection refused
	Is the server running on host "localhost" (127.0.0.1) and accepting
	TCP/IP connections on port 5432?
</code></pre>
<p>This is my workflow right now. Any ideas on what I am doing wrong?</p>
<pre><code class="lang-yaml">name: server-ci

on:
  pull_request:
      types: [opened]

env:
  DJANGO_SETTINGS_MODULE: settings_test

jobs:

  run-tests:
    name: Run tests

    runs-on: ubuntu-latest

    container:
      image: myimage/django-server:base
      credentials:
        username: ${{ secrets.DOCKERHUB_USERNAME }}
        password: ${{ secrets.DOCKERHUB_PASSWORD }}
      ports:
        - 8000:8000

    services:
      postgres:
        image: postgres
        env:
          POSTGRES_USER: postgres
          POSTGRES_PASSWORD: admin
          POSTGRES_DB: mydb
        ports:
          - 5432:5432
        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5

    env:
      POSTGRES_HOST: localhost
      POSTGRES_PORT: 5432
      POSTGRES_PASSWORD: admin
      POSTGRES_USER: postgres

    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - name: Cache dependencies
        uses: actions/cache@v2
        with:
          path: /opt/venv
          key: /opt/venv-${{ hashFiles('**/requirements.txt') }}

      - name: Install dependencies
        run: |
          python -m pip install --upgrade pip
          python -m pip install -r requirements.txt

        if: steps.cache.outputs.cache-hit != 'true'
      - name: Run tests
        run: |
          ./manage.py test --parallel --verbosity=2
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>Your workflow is now running in a container of it’s own, next to the <code>postgres</code> container. So the port mapping to the runner VM doesn’t do anything any more (because it affects the host, not Docker containers on it).</p>
<p>The job and service containers get attached to the same Docker network, so you should only need to change <code>POSTGRES_HOST</code> to <code>postgres</code> (the name of the service container) and Docker’s DNS should do the rest.</p>

**Reply by FerroEduardo** [reply]
Thanks a lot, you saved my day. I was facing a similar problem when creating/using a Postgres container as a database for my API tests

**Comment by everspader** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="189994">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>The job and service containers get attached to the same Docker network, so you should only need to change <code>POSTGRES_HOST</code> to <code>postgres</code> (the n</p>
</blockquote>
</aside>
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> Thank you so much for your help. I spent too much time looking at the wrong places!</p>
